dev/breeze/src/airflow_breeze/commands/setup_commands.py

```diff
@@ -269,6 +271,24 @@ def dict_hash(dictionary: dict[str, Any]) -> str:
     return dhash.hexdigest()
 
 
+def is_common_param(name):
+    return name == "verbose" or name == "help" or name == "dry_run"
+
+
+def validate_params_for_command(command_params, command):
+    global OPTIONS_COMMAND_MAP
+    if "params" in command_params:
+        for param in command_params["params"]:
+            name = param["name"]
+            opts = param["opts"]
+            if not is_common_param(name):
+                for opt in opts:
+                    if name not in OPTIONS_COMMAND_MAP:
+                        OPTIONS_COMMAND_MAP[opt] = [command]
+                    else:
```

Comment:
I think it's not checking what it should be checking. There are two mistakes here:

a) there is a check for `name` in the map but then opt is added. Effectively your OPTION_COMMAND_MAP contains no duplicates (and cannot have it)

b) the way it works now is that it was supposed to check if the same "short" flag is used in several commands. But this is fine and should be perfectly ok. What we want to check if a command contains more than one option with the same "short" variant.


Currentlly it checks if the same (short) option is used in more than one command

Reply:
For example you can modify (in common_options.py)

```
option_upgrade_to_newer_dependencies = click.option(
    "-u",
    "--upgrade-to-newer-dependencies",
    is_flag=True,
    help="When set, upgrade all PIP packages to latest.",
    envvar="UPGRADE_TO_NEWER_DEPENDENCIES",
)
option_upgrade_on_failure = click.option(
    "--upgrade-on-failure",
    is_flag=True,
    help="When set, attempt to run upgrade to newer dependencies when regular build fails.",
    envvar="UPGRADE_ON_FAILURE",
)
```

to be:

```
option_upgrade_to_newer_dependencies = click.option(
    "-u",
    "--upgrade-to-newer-dependencies",
    is_flag=True,
    help="When set, upgrade all PIP packages to latest.",
    envvar="UPGRADE_TO_NEWER_DEPENDENCIES",
)
option_upgrade_on_failure = click.option(
    "-u",  # <--- here added -u
    "--upgrade-on-failure",
    is_flag=True,
    help="When set, attempt to run upgrade to newer dependencies when regular build fails.",
    envvar="UPGRADE_ON_FAILURE",
)
```

and we want to detect duplicate is happening in `ci-image build` command (see the double `-u`):


<img width="1234" alt="Screenshot 2023-09-11 at 14 09 56" src="https://github.com/apache/airflow/assets/595491/7c510fc3-249a-464f-8b73-fe2cb435b5e3">